### PR TITLE
Update ImageJ2/Fiji version in the docs and improve toc layout

### PR DESCRIPTION
**Pull request by elevans**
This PR is in response to https://github.com/imagej/pyimagej/issues/271 and updates the version of ImageJ2/Fiji we use from `2.5.0` to `2.13.1`. I also applied the following changes:

- Reorganized the table of contents into sections (taking a hint from how [read the docs](https://docs.readthedocs.io/en/stable/tutorial/index.html) setup their toc).
- Replaced the blood vessel image in notebook `03-Sending-Data-to-Java` with a slice of microtubules from a HeLa cell in metaphase that I took a few years back.
- Removed references to the dummy image HACK.

### CODE REVIEW

**Review by gselzer — COMMENTED** [review]
@elevans I like all of the changes - in fact, most of my comments probably aren't related to your changes.

One final question - can you build the documentation? I ran `make docs && firefox doc/_build/html/index.html`, but it errors.

File: doc/03-Sending-Data-to-Java.ipynb [hunk 1]
```diff
@@ -176,7 +183,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "<java class 'net.imglib2.img.ImgView'>\n",
+      "<java class 'net.imagej.DefaultDataset'>\n",
```
[the remark this answers is not in the document]
Comment (elevans) [reply doc/03-Sending-Data-to-Java.ipynb:186]: Nice catch! Fixed https://github.com/imagej/pyimagej/pull/276/commits/d0232e9dc85c372a89bfb59203b2513cf494a6a1.

File: doc/06-Working-with-Images.ipynb
[the remark this answers is not in the document]
Comment (elevans) [reply doc/06-Working-with-Images.ipynb:1]: This one is okay. It's just the output of the code block that changed.

File: doc/07-Running-Macros-Scripts-and-Plugins.ipynb [hunk 2]
```diff
@@ -110,6 +110,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
+      "[INFO] stats = [[geometricMean, harmonicMean, kurtosis, max, mean, median, min, moment1AboutMean, moment2AboutMean, moment3AboutMean, moment4AboutMean, size, skewness, stdDev, sum, sumOfInverses, sumOfLogs, sumOfSquares, variance], [595.4414497753899, 577.0396027150354, 29.15435353004105, 5312.0, 620.693248, 533.0, 329.0, 3.836939868051558E-12, 46033.20548721019, 3.4518346475241706E7, 6.17816849886094E10, 62500.0, 3.4948837458522295, 214.55521907025008, 3.8793328E7, 108.31145679764538, 399331.44142784143, 2.69558321E10, 46033.94203028317]]\n",
       "           Statistic               Value\n",
       "0      geometricMean           595.44145\n",
```
[the remark this answers is not in the document]
Comment (elevans) [reply doc/07-Running-Macros-Scripts-and-Plugins.ipynb:115]: The printouts are coming from Java side of things. `ij.py.run_script()` in this example is returning the `[INFO] stats` and `[java.lang.Enum.toString]` from stdout. For now, I think we will have to unfortunately have this ugly duplicate printout until we suppress them. At least that's my understanding. Thoughts?

File: doc/07-Running-Macros-Scripts-and-Plugins.ipynb [hunk 3]
```diff
@@ -131,6 +132,13 @@
       "17      sumOfSquares       26955832100.0\n",
       "18          variance         46033.94203\n"
      ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[java.lang.Enum.toString] [INFO] stats = [[geometricMean, harmonicMean, kurtosis, max, mean, median, min, moment1AboutMean, moment2AboutMean, moment3AboutMean, moment4AboutMean, size, skewness, stdDev, sum, sumOfInverses, sumOfLogs, sumOfSquares, variance], [595.4414497753899, 577.0396027150354, 29.15435353004105, 5312.0, 620.693248, 533.0, 329.0, 3.836939868051558E-12, 46033.20548721019, 3.4518346475241706E7, 6.17816849886094E10, 62500.0, 3.4948837458522295, 214.55521907025008, 3.8793328E7, 108.31145679764538, 399331.44142784143, 2.69558321E10, 46033.94203028317]]\n"
```
[the remark this answers is not in the document]
Comment (elevans) [reply doc/07-Running-Macros-Scripts-and-Plugins.ipynb:140]: This is coming form `ij.py.run_script()`. Not sure exactly why, but we're not getting more stdout from Java in this case.

File: doc/07-Running-Macros-Scripts-and-Plugins.ipynb [hunk 4]
```diff
@@ -238,16 +246,28 @@
...
+      "[INFO] blobs.gif = net.imagej.display.DefaultDatasetView@a323a5b\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[java.lang.Enum.toString] [INFO] blobs.gif = net.imagej.display.DefaultDatasetView@a323a5b\n"
+     ]
+    },
```
[the remark this answers is not in the document]
Comment (elevans) [reply doc/07-Running-Macros-Scripts-and-Plugins.ipynb:262]: I agree!

File: doc/08-Discover-and-run-ImageJ-commands.ipynb [hunk 5]
```diff
@@ -138,22 +123,24 @@
...
       "Operating in headless mode - the IJ class will not be fully functional.\n"
      ]
     },
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "[INFO] null = img[\"test_still.tif\" (-3), 16-bit, 300x300x1x1x1]\n"
+      "[INFO] test_still.tif = net.imagej.display.DefaultDatasetView@73a8e994\n",
+      "[INFO] null = img[\"test_still.tif\" (-2), 16-bit, 300x300x1x1x1]\n"
```
[the remark this answers is not in the document]
Comment (elevans) [reply doc/08-Discover-and-run-ImageJ-commands.ipynb]: This does look related....something is certainly more verbose now.

**Comment by hinerm** [comment]
@elevans @gselzer what do you think about changing the verbosity in the affected notebooks to avoid these printouts? Is that a possibility?

**Comment by elevans** [comment]
I think this might be confusing for people and might generate erroneous bug reports. If someone sees our notebooks with the _clean_ output but theirs is duplicating they might reach out to us.

File: doc/07-Running-Macros-Scripts-and-Plugins.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (elevans) [reply doc/07-Running-Macros-Scripts-and-Plugins.ipynb:108]: The duplicate outputs have been fixed: https://github.com/imagej/pyimagej/commit/d99fcf022abf68cbb137f78025beef5396f85eb3.

File: doc/07-Running-Macros-Scripts-and-Plugins.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (elevans) [reply doc/07-Running-Macros-Scripts-and-Plugins.ipynb]: Fixed: https://github.com/imagej/pyimagej/commit/d99fcf022abf68cbb137f78025beef5396f85eb3

File: doc/07-Running-Macros-Scripts-and-Plugins.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (elevans) [reply doc/07-Running-Macros-Scripts-and-Plugins.ipynb]: Fixed: https://github.com/imagej/pyimagej/commit/d99fcf022abf68cbb137f78025beef5396f85eb3

File: doc/08-Discover-and-run-ImageJ-commands.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (elevans) [reply doc/08-Discover-and-run-ImageJ-commands.ipynb]: Fixed: https://github.com/imagej/pyimagej/commit/d99fcf022abf68cbb137f78025beef5396f85eb3

**Comment by elevans** [comment]
> @elevans I like all of the changes - in fact, most of my comments probably aren't related to your changes.
> 
> One final question - can you build the documentation? I ran `make docs && firefox doc/_build/html/index.html`, but it errors.

I wasn't able to reproduce this. This command works fine on my end. I did `make setup` first which created the `pyimagej-dev` environment and then the `make docs` command without any issues.

**Pull request merged**